core/shared/src/main/scala/org/http4s/Header.scala

```diff
@@ -179,11 +179,25 @@ object Header {
     // Required for 2.12 to convert variadic args.
     implicit def scalaCollectionSeqToRaw[H](
         h: collection.Seq[H]
+    )(implicit convert: H => ToRaw with Primitive): Header.ToRaw = scalaCollectionListToRaw(h.toList)
+
+    implicit def scalaCollectionListToRaw[H](
+      h: List[H]
     )(implicit convert: H => ToRaw with Primitive): Header.ToRaw = new Header.ToRaw {
-      val values: List[Raw] = h.toList.foldMap(v => convert(v).values)
```

Comment:
`foldMap` is really inefficient according to profiling, for whatever reason. I've replaced it with an inlined simple version below.

Reply:
Possibly related to https://github.com/http4s/http4s/pull/6712?